esp-mbedtls/src/edge_nal.rs

```diff
@@ -10,7 +10,7 @@ use crate::{Certificates, Mode, TlsError, TlsReference, TlsVersion};
 pub struct TlsAcceptor<'d, T> {
     acceptor: T,
     min_version: TlsVersion,
-    certificates: Certificates<'d>,
```

Comment:
I think the signature here and everywhere below should rather be `certificates: &'d Certificates<'d>`. This way:
- If the `Certificates` type needs to preserve a reference to `X509`, it can do it
- If it does not have to, it will essentially be `Certificates<'static>` which - due to the covariance of `'d` would be represented just fine with `&'d Certificates<'d>`